Octokit/Models/Response/StringEnum.cs

```diff
@@ -0,0 +1,156 @@
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using Octokit.Internal;
+
+namespace Octokit
+{
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    [SuppressMessage("Microsoft.Naming", "CA1711:IdentifiersShouldNotHaveIncorrectSuffix")]
+    public struct StringEnum<TEnum> : IEquatable<StringEnum<TEnum>>
+        where TEnum : struct
+    {
+        private readonly string _stringValue;
+
+        private TEnum? _parsedValue;
+
+        public StringEnum(TEnum parsedValue)
+        {
+            if (!Enum.IsDefined(typeof(TEnum), parsedValue))
+            {
+                throw GetArgumentException(parsedValue.ToString());
+            }
+
+            // Use the SimpleJsonSerializer to serialize the TEnum into the correct string according to the GitHub Api strategy
+            _stringValue = new SimpleJsonSerializer().SerializeEnum(parsedValue as Enum);
```

Comment:
I _think_ I found the issue. The enum cache dictionary is not filled with all the values of a specific enum, but only with the value we want to deserialize when calling the method.

After removing [this line](https://github.com/khellang/octokit.net/blob/d3daebc42b39ce5c07a9a93ac0657793aece241c/Octokit/Http/SimpleJsonSerializer.cs#L115), tests pass when `StringEnum` has a static instance of `SimpleJsonSerializer`. Am I missing something here and this is the desired behaviour?